src/core/lib/OrderValidator.sol

```diff
@@ -443,29 +443,12 @@ contract OrderValidator is Executor, ZoneInteraction {
                         MaxUint120
                     )
 
-                // If denominator of 1 supplied, fill entire remaining amount.
-                if eq(denominator, 1) {
-                    // Set the amount to fill to the remaining amount.
-                    numerator := sub(filledDenominator, filledNumerator)
-
-                    hasCarryOrNoNumerator := iszero(numerator)
```

Comment:
Again, explaining in my own words: this is problematic because if `filledNumerator / filledDenominator` is already `99 / 100`, and supplying a fraction of `1/1`, here `numerator` would be `1`; so `hasCarryOrNoNumerator` would be `0`, so it'd return `true` and not skip the order.